homeassistant/components/google_assistant/trait.py

```diff
@@ -841,37 +841,39 @@ def supported(domain, features, device_class, _):
 
     def sync_attributes(self):
         """Return temperature attributes for a sync request."""
+        response = {}
         domain = self.state.domain
+        attrs = self.state.attributes
         unit = self.hass.config.units.temperature_unit
+        response["temperatureUnitForUX"] = _google_temp_unit(unit)
+
         if domain == water_heater.DOMAIN:
-            min_temp = TemperatureConverter.convert(
-                self.state.attributes[water_heater.ATTR_MIN_TEMP],
-                unit,
-                UnitOfTemperature.CELSIUS,
+            min_temp = round(
+                TemperatureConverter.convert(
+                    float(attrs[water_heater.ATTR_MIN_TEMP]),
+                    unit,
+                    UnitOfTemperature.CELSIUS,
+                )
             )
-            max_temp = TemperatureConverter.convert(
-                self.state.attributes[water_heater.ATTR_MAX_TEMP],
-                unit,
-                UnitOfTemperature.CELSIUS,
+            max_temp = round(
+                TemperatureConverter.convert(
+                    float(attrs[water_heater.ATTR_MAX_TEMP]),
+                    unit,
+                    UnitOfTemperature.CELSIUS,
+                )
             )
-            if unit == UnitOfTemperature.FAHRENHEIT:
-                min_temp = round(min_temp)
-                max_temp = round(max_temp)
-            return {
-                "temperatureUnitForUX": _google_temp_unit(unit),
-                "temperatureRange": {
-                    "minThresholdCelsius": min_temp,
-                    "maxThresholdCelsius": max_temp,
-                },
+            response["temperatureRange"] = {
+                "minThresholdCelsius": min_temp,
+                "maxThresholdCelsius": max_temp,
             }
-        return {
-            "temperatureUnitForUX": _google_temp_unit(unit),
-            "queryOnlyTemperatureControl": True,
-            "temperatureRange": {
+        else:
+            response["queryOnlyTemperatureControl"] = True
+            response["temperatureRange"] = {
                 "minThresholdCelsius": -100,
                 "maxThresholdCelsius": 100,
```

Comment:
The Celsius range seems unrealistic. 0 deg Celsius is the freeze point of water. Under pressure the boiling point of water could become just above 100 deg Celsius, but the 100 deg limit seems fine to me.
I would say the lower limit should be 0 at least.

Reply:
Sorry for the delay, I've been distracted by a comp-sci degree I'm attempting to finish.
I agree `-100` is ridiculous. I wouldn't know what a sufficient lower bound would be so I think `0` is safe.

Reply:
Agree, below `0` Celcius we are heating ice at normal pressure